api/middleware/user.go

```diff
@@ -0,0 +1,34 @@
+package middleware
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/odpf/columbus/api/handlers"
+	"github.com/odpf/columbus/user"
+	"github.com/odpf/salt/log"
+)
+
+// ValidateUser middleware will propagate a valid user ID as string
+// within request context
+// use `user.FromContext` function to get the user ID string
+func ValidateUser(logger log.Logger, userSvc *user.Service) mux.MiddlewareFunc {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+			userID, err := userSvc.ValidateWithHeader(r.Context(), r.Header)
+			if err != nil {
+				logger.Warn("error validating user", "error", err)
+				if errors.Is(errors.Unwrap(err), user.ErrNoUserInformation) {
+					handlers.WriteJSONError(rw, http.StatusBadRequest, err.Error())
```

Comment:
are we returning `http.StatusBadRequest` because the user info is used for enriching the data/record instead of authorization?

Reply:
yes, this is not really an authorization, unless we don't do a `real` authorization check, i don't think we should return 401